File: 코딩테스트 고득점 Kit/DFSBFS/여행경로/여행경로.py
def solution(tickets):
    answer = []
    visited = [False] * len(tickets)
    tickets.sort()

    def DFS(visited, cur, path):
        nonlocal answer
        if answer:
            return

        if not answer and len(path) == len(tickets)+1:
            answer.extend(path)
            return

        for i in range(len(tickets)):
            start = tickets[i][0]
            end = tickets[i][1]

            if not visited[i] and cur == start:
                visited[i] = True
                path.append(end)

                DFS(visited, end, path)

                path.pop()
                visited[i] = False
        return

    DFS(visited, "ICN", ["ICN"])
    return answer

File: 코딩테스트 고득점 Kit/DFSBFS/여행경로/여행경로.swift
import Foundation

func solution(_ tickets:[[String]]) -> [String] {
    var answer:[String] = [] 
    var visited = Array(repeating: false, count: tickets.count)
    var tickets = tickets.sorted { $0[1] < $1[1] }

    func DFS(_ cur:String, _ path: [String]) {
        if answer.count != 0 {
            return
        }
        
        if  answer.count == 0 && path.count == tickets.count + 1 {
            answer = path
            return 
        }
        
        var path = path
        
        for i in 0..<tickets.count {
            let start = tickets[i][0]
            let next = tickets[i][1]
            
            if !visited[i] && cur == start {
                visited[i] = true
                path.append(next) 

                DFS(next, path) 

                path.removeLast()
                visited[i] = false
            }
        }
    }
    
    DFS("ICN", ["ICN"])

    return answer
}